health-monitors/gpu-health-monitor/gpu_health_monitor/dcgm_watcher/dcgm.py

```
@@ -272,9 +374,41 @@ def _initialize_dcgm_monitoring(self, dcgm_handle: pydcgm.DcgmHandle) -> tuple:
+                )
+            elif self._thermal_margin_enabled:
+                log.warning("GpuThermalMarginWatch enabled but no metadata reader configured; skipping field watch")
+
             return dcgm_group, gpu_ids, gpu_serials
         except Exception as e:
             log.warning(f"DCGM monitoring initialization failed, rolling back group: {e}")
+            if self._field_group is not None:
+                try:
+                    dcgm_group.samples.UnwatchFields(self._field_group)
+                    self._field_group.Delete()
+                except Exception as unwatch_err:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)